lhotse/cut.py

```diff
@@ -760,6 +760,50 @@ class MonoCut(Cut):
     # For the cases that the model was trained by raw audio instead of features
     recording: Optional[Recording] = None
 
+    # Store anything else the user might want.
+    custom: Optional[Dict[str, Any]] = None
+
+    def __setattr__(self, key: str, value: Any):
+        if key in self.__dataclass_fields__:
+            super().__setattr__(key, value)
+        else:
+            custom = ifnone(self.custom, {})
+            custom[key] = value
+            self.custom = custom
+
+    def __getattr__(self, item: str) -> Any:
+        """
+        __getattr__ only gets called for attributes that don't already exist.
+        We use it to look up the ``custom`` field: when it's None or empty,
+        we'll just raise AttributeError as usual.
+        If ``item`` is found in ``custom``, we'll return ``custom[item]``.
+        If ``item`` starts with "load_", we'll assume the name of the relevant
+        attribute comes after that, and that value of that field is of type
+        Array or TemporalArray. We'll return it's ``load`` method to call by the user.
+        """
+        from lhotse.array import Array, TemporalArray
+
+        custom = self.custom
+        if custom is None:
+            raise AttributeError()
```

Comment:
Shall we also display the name of the accessed attribute in the exception?
Just for better diagnostics.